server/models/procedures.sql

```diff
@@ -18,6 +18,235 @@ BEGIN
 END
 $$
 
+CREATE PROCEDURE StageInvoice(
+  IN is_distributable TINYINT(1),
+  IN date DATETIME,
+  IN cost DECIMAL(19, 4) UNSIGNED,
+  IN description TEXT,
+  IN service_id SMALLINT(5) UNSIGNED,
+  IN debtor_uuid BINARY(16),
+  IN project_id SMALLINT(5),
+  IN user_id SMALLINT(5),
+  IN uuid BINARY(16)
+)
+BEGIN 
+  DECLARE `no_invoice_stage` TINYINT(1) DEFAULT 0;
+  DECLARE CONTINUE HANDLER FOR SQLSTATE '42S02' SET `no_invoice_stage` = 1;
+  SELECT NULL FROM `stage_invoice` LIMIT 0;
+
+  IF (`no_invoice_stage` = 1) THEN 
```

Comment:
In this case the error is handled locally - `stage_invoice` will not exist for the first invoice write request on any given MySQL connection. 
**If it does not exist**
* create temporary table stage_invoice 
* select data provided into table 

**If it does exist**
* select data provided into table